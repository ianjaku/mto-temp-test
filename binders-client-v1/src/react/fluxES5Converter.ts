/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
