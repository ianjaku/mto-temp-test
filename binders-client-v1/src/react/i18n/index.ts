import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
