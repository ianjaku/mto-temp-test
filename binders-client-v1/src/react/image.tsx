/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
