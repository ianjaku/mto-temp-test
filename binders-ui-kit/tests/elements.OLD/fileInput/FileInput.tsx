import * as React from "react";
import FileInput from "../../../src/elements/fileselector";
import { create } from "react-test-renderer";
import { mount } from "enzyme";

const TEXT = "Select .csv file";
const ACCEPTED_LIST = [".csv", ".jpg", "image/*"];
const NAME = "finput";

const createFileInput = () => <FileInput name="finput" />;
const createFileInputCustomButtonText = () => (
    <FileInput name={NAME} buttonText={TEXT} accept={ACCEPTED_LIST} />
);
const createFileInputWithCallback = callback => <FileInput name="finput" onChange={callback} />;

// most preferred
describe("File Input", () => {
    test("Snapshot default file input", () => {
        const input = create(createFileInput()).toJSON();
        expect(input).toMatchSnapshot();
    });

    test("Name should be correct", () => {
        const element = mount(createFileInput());
        const comp = element.find(".file-input > label > input");
        expect(comp.prop("name")).toEqual(NAME);
        expect(comp.prop("type")).toEqual("file");
        expect(comp.prop("accept")).toEqual("*");
    });

    test("Accept can be modified", () => {
        const element = mount(createFileInputCustomButtonText());
        const comp = element.find(".file-input > label > input");
        expect(comp.prop("accept")).toEqual(ACCEPTED_LIST.join(","));
    });

    test("Button text can be set", () => {
        const element = mount(createFileInputCustomButtonText());
        const comp = element.find(".file-input > label");
        expect(comp.text()).toEqual(TEXT);
    });

    test("Callback works", () => {
        const callback = jest.fn();
        const element = mount(createFileInputWithCallback(callback));
        const comp = element.find(".file-input > label > input");

        // create file and test
        const fileContents = "file contents";
        const file = new Blob([fileContents], { type: "text/plain" });
        comp.simulate("change", { target: { files: [file] } });

        expect(callback.mock.calls.length).toEqual(1);
    });
});
