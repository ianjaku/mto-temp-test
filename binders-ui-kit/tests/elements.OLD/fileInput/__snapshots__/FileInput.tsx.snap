// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Input Snapshot default file input 1`] = `
<div
  className="file-input"
>
  <label
    htmlFor="finput"
  >
    Choose a file
    <input
      accept="*"
      multiple={false}
      name="finput"
      onChange={[Function]}
      type="file"
    />
  </label>
  <span
    className="files"
  >
    No file chosen
  </span>
</div>
`;
